onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -1862,6 +1862,64 @@ namespace OperatorHelper
         return { std::move(outputShape) };
     }
 
+    void Col2ImHelper::Initialize(
+        const IKernelInformationAdapter& kernelInformation,
+        const IShapeInformationAdapter& shapeInformation)
+    {
+        std::vector<int> tensor;
+        ReadCpuLocalTensorIntoInt32(kernelInformation.GetConstantInputTensor(1), /*out*/ tensor);
+        m_imageShape.resize(tensor.size());
+        std::transform(tensor.begin(), tensor.end(), m_imageShape.begin(), [](auto& x){ return (uint32_t)x; });
+        ReadCpuLocalTensorIntoInt32(kernelInformation.GetConstantInputTensor(2), /*out*/ tensor);
+        m_blockShape.resize(tensor.size());
+        std::transform(tensor.begin(), tensor.end(), m_blockShape.begin(), [](auto& x){ return (uint32_t)x; });
+         
+        const uint32_t dimCount = gsl::narrow_cast<uint32_t>(m_blockShape.size());
+        m_dilations = {dimCount, 1};
+        m_pads = {dimCount * 2, 0};
+        m_strides = {dimCount, 1};
+
+        if (kernelInformation.HasAttribute(AttrName::Dilations, MLOperatorAttributeType::IntArray))
+        {
+            tensor = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Dilations);
+            m_dilations.resize(tensor.size());
+            std::transform(tensor.begin(), tensor.end(), m_dilations.begin(), [](auto& x){ return (uint32_t)x; });
+            ML_CHECK_VALID_ARGUMENT(m_dilations.size() == dimCount);
+        }
+
+        if (kernelInformation.HasAttribute(AttrName::Pads, MLOperatorAttributeType::IntArray))
+        {
+            tensor = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Pads);
+            m_pads.resize(tensor.size());
+            std::transform(tensor.begin(), tensor.end(), m_pads.begin(), [](auto& x){ return (uint32_t)x; });
+            ML_CHECK_VALID_ARGUMENT(m_pads.size() == dimCount * 2);
+        }
+
+        if (kernelInformation.HasAttribute(AttrName::Strides, MLOperatorAttributeType::IntArray))
+        {
+            tensor = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Strides);
+            m_strides.resize(tensor.size());
+            std::transform(tensor.begin(), tensor.end(), m_strides.begin(), [](auto& x){ return (uint32_t)x; });
```

Comment:
This pattern is repeated quite a few times. We already have a similar helper `DowncastDimensions`. Maybe we could just add an overload for `int32_t`. Then it becomes:

```
tensor = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Strides);
ML_CHECK_VALID_ARGUMENT(tensor.size() == dimCount);
DowncastDimensions(tensor, /*out*/ m_strides);
```